src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -505,7 +505,13 @@ class Address {
   }
 
   bool uses(Register reg) const {
-    return base() == reg || index() == reg;
+    switch (_mode) {
+    case literal:
+    case no_mode:
+      return false;
+    default:
+      return base() == reg || index() == reg;
+    }
```

Comment:
It'd be more robust if you enumerated all of the possibilities here, and used `ShouldNotReachHere()` on the default case.

Reply:
That's a good suggestion. I've made this change in the latest version.